doc/spec/openapi.yaml

```diff
@@ -0,0 +1,441 @@
+openapi: 3.0.0
```

Comment:
My understanding is that `openapi.yaml` is manually edited and `redoc-static.html` is built from that. Are there methods of verifying the OpenAPI spec against a live API, so we can ensure they don't go out of sync?

Or maybe I should add an integration test into Kosa's normal test suite to do this?

Reply:
indeed the reitit library has https://github.com/metosin/reitit/blob/master/doc/ring/openapi.md support for openapi generation from code. It would need macros to be added though.

Reply:
Very cool! I've added an issue to track this here: https://github.com/pariyatti/kosa/issues/95

Ignoring that for now, I think this PR is good to go. :ok_hand: